core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1032,7 +1056,12 @@ public MetaData build() {
                 }
             }
             aliasAndIndexLookup = Collections.unmodifiableSortedMap(aliasAndIndexLookup);
-            return new MetaData(clusterUUID, version, transientSettings, persistentSettings, indices.build(), templates.build(), customs.build(), allIndices, allOpenIndices, allClosedIndices, aliasAndIndexLookup);
+            if (customs.get(IndexGraveyard.TYPE) == null) {
+                // not part of the customs map, so add a fresh entry
+                customs.put(IndexGraveyard.TYPE, IndexGraveyard.builder().build());
```

Comment:
Reason I didn't do it like this is because all of the constructors for `IndexGraveyard` are private.  I can change that, but in some ways it defeats the purpose of the Builder.  What do you think?